Add release notes for PR 85 and PR 89 (#112)
frequenz-floss/frequenz-api-weather: RELEASE_NOTES.md

## New Features

> - Extend the client to iterate over historical weather forecast data.
> 
> - Convert a page of historical location forecast data into a flattened numpy array for easier use.

Reviewer: Release notes need to have a different tone than commit messages, I think, because they are bigger picture.  I've made some suggestions, do they still match the changes from the PRs?

```suggestion
- The client has been extended to iterate over historical weather forecast data.

- Historical location forecast data pages are now exposed as a flattened numpy array for easier use.
```
